src/index.ts

```diff
@@ -29,7 +29,10 @@ export const cx = <T extends CxOptions>(...classes: T): CxReturn =>
 type ConfigSchema = Record<string, Record<string, ClassValue>>;
 
 type ConfigVariants<T extends ConfigSchema> = {
-  [Variant in keyof T]?: StringToBoolean<keyof T[Variant]> | null;
+  [Variant in keyof T]?:
+    | StringToBoolean<keyof T[Variant]>
+    | StringToBoolean<keyof T[Variant]>[]
+    | null;
```

Comment:
I think we need to find a way to separate this into "internal config" vs "external config". At the moment this means that any consumer could erroneously pass an array of multiple variants to the (rather than specifically to `compoundVariants` only)

Before this addition:

<img width="736" alt="CleanShot 2022-11-28 at 10 35 29@2x" src="https://user-images.githubusercontent.com/7349341/204165732-2712ca57-937d-474c-91a5-d569c26974d7.png">

After this addition:

<img width="530" alt="CleanShot 2022-11-28 at 10 36 02@2x" src="https://user-images.githubusercontent.com/7349341/204165744-e932b2c2-81a3-427a-9066-f5a4ca090a07.png">

Reply:
It's times like these when I wish I was a typescript wizard :(

I managed to get help in creating a type that accepts a specific combination of array values regardless of their order, but couldn't move the ball further.

Link [here](https://www.typescriptlang.org/play?#code/MYewdgzgLgBAhsYBTADlJATAwiAtgIxBgF4YBtAcjgoBoYL9b7gKBdeCGUSKAKF4D0AmNBAAbJGBgAzAE54YACySykSqFBQQAXEIDmASyiKArvgB0oXANwHg8iCGlQBAFQCeKJAGV7BtAIGEBAmSBACAIwAzABMAJwAHADEQSFIVriSUAC0AGxxMQm5ACwADAkA7LxQnmquJigSAKpgBuAAPE0wSAAe6GAYnNCyBmB6dABK3X2Sg-Bg7mTspEsAfCQwAN68MLvk3jCjME2s2jAAoj3AYiYYSJ103uu9-XNgSABuKjAA-OTmAImj3YZ3qjSQLTaYHal2ut3uTUeqzoZAB5iBMG8rFWAG5eABfMgnPH8GpeGAAayQ7k4pDBzVaHTJSCc8EQqHQ2DwhDIYBMBBU2JJ0hMYGAUChMhAIHarmmryGUBGY1WAAoUHBZHBcGcqTSAJRbfH8XjSaWqyjUOgMNiGoQwADyAGlTebKIxrdRWHbhM7XSALVQmDbrSxvTB7R4vOdZPJZDBsjAAETAJOHTgvLX+wNen0wKNIGNxhPJ-BpoIwWzBUZ6fhmgOW4Me+iMcP2gByRGZ3VjIHjuhgQA) - I hope it helps whoever knows enough typescript to get this done!

Reply:
Think I've figured it out 👌